enterprise/models.py

```diff
@@ -2511,12 +2513,31 @@ class DefaultEnterpriseEnrollmentIntention(TimeStampedModel, SoftDeletableModel)
     )
     history = HistoricalRecords()
 
+    def get_cached_content_metadata(self):
+        """
+        Retrieves the cached content metadata for the instance content_key
+        Determines catalog from the catalogs associated to the enterprise customer
+        """
+        return get_and_cache_catalog_content_metadata(
+                enterprise_customer_uuid=self.enterprise_customer.uuid,
+                content_key=self.content_key,
+            )
+
+
     @cached_property
```

Comment:
[consideration, sanity check] Upon further reflection, I'm not actually sure we'd want to have this property use the `@cached_property` decorator vs. the normal `@property` decorator.

Rationale: I believe `@cached_property` will persist as long as the model instance does, unless manually invalidated or the model instance is updated (e.g., explicitly re-saved). However, the `current_course_run` ideally should be able to change over time based on the configured `content_key` (i.e., when it's a top-level course, relying on the advertised run) without needing to manually re-save instances or manually invalidate the cached property value.

I think this is mitigated by the fact that we'd be utilizing cached (`TieredCache`) content metadata here regardless, but will pull in fresh data once the TTL elapses.